### ISSUE TITLE
Postgres table-valued stored function: passing input parameter returns SQLDialectNotSupportedException on invocation

### ISSUE THREAD

**Issue by uncbailey1** [jOOQ/jOOQ#8612]
I have a stored function which needs to take a row of data from a table as an input parameter, but when I actually call this procedure with an instance of the generated record class from this table, i get a SQLDialectNotSupportedException.  I have a generated table and tablerecord class that has been working fine in general for quite a while, but when we've never tried to pass in this type to a procedure.

The following is a contrived example of the procedure that just takes in a row from the table and returns a simple string.

```sql
CREATE FUNCTION do_some_stuff(in_param myTable) RETURNS TEXT AS $$
BEGIN
    RETURN 'test';
END $$ LANGUAGE plpgsql;
```

The java code would then do the following:

```java
DoSomeStuff doSomeStuff = new DoSomeStuff();
doSomeStuff.attach(create.configuration());
doSomeStuff.setIn_param(new MyTableRecord(parm1, param2));
doSomeStuff.execute();
return doSomeStuff.getReturnValue();
```

When it tries to run, I get the unsupported exception on the MyTableRecord.  My guess is that I need to apply some configuration setting to get this type recognized as part of the Postgres Dialect, but I don't see anything that looks like it would be appropriate.

org.jooq.exception.SQLDialectNotSupportedException: Type class jooq.internal.tables.records.MyTableRecord is not supported in dialect POSTGRES

### Versions:
- jOOQ: 3.7.3
- Java: build 1.8.0_141-b16
- Database (include vendor): Postgreql 10.0
- OS: Centox 7
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for reporting. That looks more like a bug than an API misuse. Will investigate soon.

**Comment by uncbailey1** [comment]
FWIW, if I manually define a type in my script which matches the rows in the table above (i.e., it mirrors the myTableRecord generated class), then I can get this to work as it generates a UDT class which I can pass in.  The UDT seems to be the key to getting away from the SQLDialectNotSupportedException.  I did the following:

This type would mirror the table definition and the generated MyTableRecord class.
CREATE TYPE my_duplicate_table_row AS (
    id text,
    foo text
)

Change the stored procedure to take the new UDT
CREATE FUNCTION do_some_stuff(in_param my_duplicate_table_row ) RETURNS TEXT AS $$ BEGIN RETURN 'test'; END $$ LANGUAGE plpgsql;

**Comment by lukaseder** [comment]
Related: 

- https://github.com/jOOQ/jOOQ/issues/3942.

Duplicate (implemented in jOOQ 3.11):

- https://github.com/jOOQ/jOOQ/issues/7174

In jOOQ 3.19, I can't reproduce any issues here. A test function like this:

```sql
CREATE OR REPLACE FUNCTION f_search_book_by_record(p_book t_book)
  RETURNS t_book AS
$BODY$
SELECT * FROM t_book
WHERE (LOWER(title) LIKE LOWER('%' || ($1).title || '%'))
ORDER BY ID
LIMIT 1;
$BODY$
  LANGUAGE sql VOLATILE
  COST 100;
```

Can be called without problems, producing this kind of SQL:

```sql
select "public"."f_search_book_by_record"("p_book" := cast(row(1, 1, null, null, '1984', 1948, 1, '...', null, cast('ORDERED' as "public"."u_book_status"), 1, timestamp '2010-01-01 00:00:00.0') as "public"."t_book"))
```
